### PR TITLE
fix(funnels): Resolve disappearing funnel bars

### LINKED ISSUE #11872: Funnel bars not showing for first funnel stage on cohort breakdown for some teams
## Bug description

funnel bars are not appearing as expected. This persists even if I remove cohort breakdowns

image

not sure exactly how to repro this state, but on 2299 you can see it [here](https://app.posthog.com/insights/KbalvGPj). I've tried just breaking down by 6 + cohorts, but no luck there. There are some 4xx responses from `/decide` I see, but unsure if related.

## How to reproduce

1. ???
2.
3.

## Environment

- [ ] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: please provide

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Closes #11872.
Steps to reproduce the issue in question:

1. Create a new funnel.
2. Add two steps to it.
4. Add any breakdown to the funnel.
5. Uncheck one or more breakdown values from being in the funnel series table.
6. Add a new step to the funnel.

This results in funnel bars being still properly hidden for the unchecked breakdown values in the existing steps BUT unexpectedly shown in the new step.

## Changes

This simplifies funnel series keying so that visibility preferences don't have to care about the number of steps (which is never relevant), only about the breakdown values.
~~There's a dragon here: Existing saved series visibility preferences will be reset for all funnels. I haven't found a way to preserve it without a complex and unreliable migration though, because the existing `getVisibilityIndex()` scheme has been fundamentally misaligned with how the feature of unchecking breakdown values work (e.g. it was also broken for funnels where the first step uses an action instead of an event).~~ See comment below.

### CODE REVIEW

File: frontend/src/scenes/funnels/funnelLogic.ts [hunk 1]
```diff
@@ -1198,26 +1198,10 @@ export const funnelLogic = kea<funnelLogicType>({
...
-            // hide all but the first five breakdowns for each step
-            values.steps?.forEach((step) => {
-                values.flattenedStepsByBreakdown
-                    ?.filter((s) => !!s.breakdown)
-                    ?.slice(5)
-                    .forEach((b) => {
-                        actions.setHiddenById({ [getVisibilityIndex(step, b.breakdown_value)]: true })
-                    })
-            })
-        },
```
Comment (Twixes) [inline frontend/src/scenes/funnels/funnelLogic.ts:1214]: This behavior has been pretty annoying, in that we were changing visibility preferences from under the user. Looks like it was added in #12113, do you remember what was the motivation @alexkim205?

**Comment by Twixes** [comment]
Thinking about this more, we should be minimizing the disruption to user workflows. Users should not go into PostHog and see that an insight changed without their intervention, so I added some code to the `hiddenLegendKeys` selector in `insightLogic` which adds backwards compatibility with the part of the old scheme that did work before.

File: frontend/src/scenes/funnels/funnelLogic.ts (hunk 1 above)
Comment (Twixes) [reply frontend/src/scenes/funnels/funnelLogic.ts:1214]: Oh LOL meant #5733

**Review by pauldambra — APPROVED** [review]
:shipit:

**Pull request merged**